### DISCUSSION TITLE
How to export Cython binary wheels from Tox ?

### DISCUSSION THREAD

**Discussion by bsrdjan** [tox-dev/tox#2997] [Q&A] [answered]
In my Cython project, wheels must be generated for each Python version and platform and attached to Github release, like [here](https://github.com/SAP/PyRFC/releases/tag/v2.8.3)

I configured Tox to build these wheels and test them with different Python versions and all works fine.

I could  not figure out how to "export" these wheels from `.tox` folder (suppose the wheels are there), so they can be uploaded to  to GH release? Short shell script could activate pyenvs and build wheels at expected location but using Tox would be more elegant, if possible.

The question/requirement is common for Cython projects I suppose.

**Comment by gaborbernat** [comment, marked as answer]
https://cibuildwheel.readthedocs.io/en/stable is the PyPA recommended solution. For tox you would need to use the build tool https://github.com/pypa/build and some dedicated build environments to achieve similar.

**Reply by bsrdjan** [reply]
Thank you @gaborbernat. `python -m build --no-isolation` did the trick, wheels are created at requested location.

**Reply by gaborbernat** [reply]
`--no-isolation` is very much not recommended 😊

**Reply by bsrdjan** [reply]
right, I ended up with simple solution, without `---no-isolation`

```shell
python -m build --wheel --outdir dist
```
